ag/review.go

```diff
@@ -1,53 +1,31 @@
 package ag
 
-import (
-	"encoding/json"
-	"regexp"
-	"strings"
-)
+import "log"
 
-// MarshalReviewString generates a slice of JSON strings to store in the database
-func (r *Review) MarshalReviewString() error {
-	str := make([]string, 0)
-	for _, bm := range r.Benchmarks {
-		b, err := json.Marshal(bm)
-		if err != nil {
-			return err
-		}
-		str = append(str, string(b))
-	}
-	r.Review = strings.Join(str, "; ")
-	return nil
-}
-
-var re = regexp.MustCompile(`("\w+"):"(\d+)",`)
+// CalculateScore returns total score for the review. If grading criteria give
+// a predefined amount of points (not necessary summing up to 100), returns the sum of such points.
+// Otherwise, gives each criterion an equal weight and sets max score to 100
+func (r *Review) CalculateScore() {
```

Comment:
- We should have tests for this method.
- Also, suggest renaming it to `ComputeScore`.
- Remove blank line after method signature.
- The method comment appears to return the score rather than assign the value to the `Review` object. What about the comment below.

```golang
// ComputeScore computes the total score for the review and assigns it to r.
// If the grading criteria have predefined points, the score is the sum of these points.
// Otherwise, each criterion is given equal weight, such that the max score is 100.
```

Reply:
Renamed, test added.